src/taipy/_cli/_migrate.py

```diff
@@ -0,0 +1,483 @@
+
+
+def __update_parent_ids(entity: Dict, data: Dict) -> Dict:
+    # parent_id was replaced by parent_ids
+    _ = entity.pop("parent_id", None)
+
+    # parent_ids was not present in 2.0, need to be search for in tasks
+    parent_ids = entity.get("parent_ids", [])
+    if not parent_ids:
+        parent_ids = __search_parent_ids(entity["id"], data)
+    entity["parent_ids"] = parent_ids
+
+    return entity
+
+
+def __update_config_parent_ids(id: str, entity: Dict, entity_type: str, config: Dict) -> Dict:
+    # parent_id was replaced by parent_ids
+    _ = entity.pop("parent_id", None)
+
+    # parent_ids was not present in 2.0, need to be search for in tasks
+    parent_ids = entity.get("parent_ids", [])
+
+    if not parent_ids:
+        parent_ids = __search_parent_config(id, config, entity_type)
+    entity["parent_ids"] = parent_ids
+
+    return entity
+
+
+def __search_parent_ids(entity_id: str, data: Dict) -> List:
+    parents = []
+    entity_type = entity_id.split("_", 1)[0]
+
+    for _id, entity_data in data.items():
+        entity_data = entity_data["data"]
+        if entity_type == "DATANODE" and "TASK" in _id:
+            if entity_id in entity_data["input_ids"] or entity_id in entity_data["output_ids"]:
+                parents.append(_id)
+
+        if entity_type == "TASK" and "SCENARIO" in _id:
+            if entity_id in entity_data["tasks"]:
+                parents.append(_id)
+    return parents
+
+
+def __search_parent_config(entity_id: str, config: Dict, entity_type: str) -> List:
+    parents = []
+    possible_parents = "TASK" if entity_type == "DATA_NODE" else "SCENARIO"
+    data = config[possible_parents]
+
+    for _id, entity_data in data.items():
+        section_id = f"{entity_id}:SECTION"
+        if entity_type == "DATANODE" and possible_parents == "TASK":
+            if section_id in entity_data["input_ids"] or section_id in entity_data["output_ids"]:
+                parents.append(section_id)
+
+        if entity_type == "TASK" and possible_parents == "SCENARIO":
+            if section_id in entity_data["tasks"]:
+                parents.append(section_id)
+    return parents
+
+
+def __fetch_tasks_from_pipelines(pipelines: List, data: Dict) -> List:
+    tasks = []
+    for pipeline in pipelines:
+        pipeline_data = data[pipeline]["data"]
+        tasks.extend(pipeline_data["tasks"])
+    return tasks
+
+
+def _migrate_scenario(scenario: Dict, data: Dict) -> Dict:
+    # pipelines were replaced by tasks
+    scenario["tasks"] = __fetch_tasks_from_pipelines(scenario["pipelines"], data)
+
+    # pipeline attribute not removed in 3.0
+    scenario["pipelines"] = None
+
+    # additional_data_nodes attribute added
+    scenario["additional_data_nodes"] = None
+
+    return scenario
+
+
+def _migrate_task(task: Dict) -> Dict:
+    # owner_id was not present in 2.0
+    if not task.get("owner_id"):
+        task["owner_id"] = task["parent_ids"][0] if task["parent_ids"] else None
```

Comment:
```suggestion
        task["owner_id"] = task["parent_id"][0] if task["parent_id"] else None
```
I believe we need to migrate `parent_id`, not `parent_ids`, to `owner_id`.
Also, if the `parent_id` exists, we need to delete it.

Reply:
well spotted.